opengrok-web/src/main/java/org/opengrok/web/api/v1/controller/ProjectsController.java

```diff
@@ -135,35 +135,55 @@ public Response addProject(final String projectName) {
     @DELETE
     @Path("/{project}")
     public void deleteProject(@PathParam("project") final String projectName)
-            throws IOException, HistoryException {
+            throws HistoryException {
 
-        Project proj = env.getProjects().get(projectName);
-        if (proj == null) {
+        Project project = env.getProjects().get(projectName);
+        if (project == null) {
             throw new IllegalStateException("cannot get project \"" + projectName + "\"");
         }
 
+        project.setIndexed(false);
```

Comment:
this is not necessary if you consider you're removing it

Reply:
This is to avoid exceptions when performing e.g. all project search.

Reply:
You mean at the point between this line and the actual removal from the project map?

Reply:
yep